app/uk/gov/hmrc/bindingtariffclassification/connector/BankHolidaysConnector.scala

```diff
@@ -16,43 +16,46 @@
 
 package uk.gov.hmrc.bindingtariffclassification.connector
 
+import play.api.Logging
 import play.api.libs.json.Json
-import uk.gov.hmrc.bindingtariffclassification.common.Logging
 import uk.gov.hmrc.bindingtariffclassification.config.AppConfig
-import uk.gov.hmrc.bindingtariffclassification.http.ProxyHttpClient
 import uk.gov.hmrc.bindingtariffclassification.metrics.HasMetrics
 import uk.gov.hmrc.bindingtariffclassification.model.BankHolidaysResponse
 import uk.gov.hmrc.bindingtariffclassification.model.RESTFormatters.formatBankHolidaysResponse
-import uk.gov.hmrc.http.HeaderCarrier
-import uk.gov.hmrc.play.bootstrap.metrics.Metrics
+import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.http.{HeaderCarrier, StringContextOps}
 
 import java.nio.charset.StandardCharsets
 import java.time.LocalDate
-import javax.inject.{Inject, Singleton}
+import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 import scala.io.Source
 
-@Singleton
-class BankHolidaysConnector @Inject() (appConfig: AppConfig, http: ProxyHttpClient, val metrics: Metrics)(implicit
+class BankHolidaysConnector @Inject() (
+  appConfig: AppConfig,
+  http: HttpClientV2,
+  hasMetrics: HasMetrics
+)(implicit
   executionContext: ExecutionContext
-) extends Logging
-    with HasMetrics {
+) extends Logging {
+
   def get()(implicit headerCarrier: HeaderCarrier): Future[Set[LocalDate]] =
-    withMetricsTimerAsync("get-bank-holidays") { _ =>
-      http
-        .GET[BankHolidaysResponse](appConfig.bankHolidaysUrl)
-        .recover(withResourcesFile)
-        .map(_.`england-and-wales`.events.map(_.date).toSet)
-    }
+    hasMetrics
+      .withMetricsTimerAsync("get-bank-holidays") { _: hasMetrics.MetricsTimer =>
+        http
+          .get(url"${appConfig.bankHolidaysUrl}")
+          .execute[BankHolidaysResponse]
+      }
+      .map(_.`england-and-wales`.events.map(_.date).toSet)
+      .recover(withResourcesFile)
```

Comment:
Seems there were a lot of things changed on this PR so this has to be thoroughly tested before deploying to production. Using this as an example as this will not work. This should use proxy as it did previously and has it configured for bank holidays [here](https://github.com/hmrc/app-config-production/blob/main/binding-tariff-classification.yaml#L61-L67) in this backend via this [SUP ticket](https://jira.tools.tax.service.gov.uk/browse/SUP-12834). So it should be using [this](https://github.com/hmrc/http-verbs?tab=readme-ov-file#using-proxy) instead. This is just one issue I have spotted looking at a glance.